ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go

```diff
@@ -1471,20 +1504,81 @@ func (h *FaultHandler) startNetworkPacketLossFaultForInterface(
 		return err
 	}
 	logger.Info("Command execution completed", logger.Fields{
-		field.CommandString:    tcAddQdiscLossCommandComposed,
+		field.CommandString:    tcAddQdiscSecondLayerCommandComposed,
 		field.CommandOutput:    string(cmdOutput[:]),
 		field.NetworkInterface: interfaceName,
 	})
-	// After creating the queueing discipline, create filters to associate the IPs in the request with the handle.
-	// First redirect the allowlisted ip addresses to band 1:3 where is no network impairments.
-	if err := h.addIPAddressesToFilter(ctx, request.SourcesToFilter, taskMetadata, nsenterPrefix, tcAllowlistIPCommandString, interfaceName); err != nil {
-		return err
-	}
-	// After processing the allowlisted ips, associate the ip addresses in Sources with the qdisc.
-	if err := h.addIPAddressesToFilter(ctx, request.Sources, taskMetadata, nsenterPrefix, tcAddFilterForIPCommandString, interfaceName); err != nil {
-		return err
+
+	// Add the third qdisc layer, one 10 band qdisc per each 2nd layer band.
+	for i := range tcLayerBands {
+		childHandle := fmt.Sprintf("%d:", 100+i) // 100:, 101:, 102:, ... 109:
+		parent := fmt.Sprintf("10:%d", i+1)      // 10:1, 10:2, ... 10:a
+		// tc uses hexadecimal for the ids so we need to change to 'a' when reaching 10.
+		if i == 9 {
+			parent = "10:a"
+		}
+		// <nsenterPrefix> tc qdisc add dev %s parent %s handle %s prio bands 10 priomap 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2
+		tcAddQdiscThirdLayerCommandComposed := nsenterPrefix + fmt.Sprintf(tcAddQdiscChildCommandString, interfaceName, parent, childHandle)
+		cmdList = strings.Split(tcAddQdiscThirdLayerCommandComposed, " ")
+		cmdOutput, err = h.runExecCommand(ctx, cmdList)
+		if err != nil {
+			logger.Error("Command execution failed", logger.Fields{
+				field.CommandString:    tcAddQdiscThirdLayerCommandComposed,
+				field.Error:            err,
+				field.CommandOutput:    string(cmdOutput[:]),
+				field.TaskARN:          taskMetadata.TaskARN,
+				field.NetworkInterface: interfaceName,
+			})
+			return err
+		}
+		logger.Info("Command execution completed", logger.Fields{
+			field.CommandString:    tcAddQdiscThirdLayerCommandComposed,
+			field.CommandOutput:    string(cmdOutput[:]),
+			field.NetworkInterface: interfaceName,
+		})
 	}
+	return nil
+}
 
+func (h *FaultHandler) addNetemFaultToInterfaceFlows(
+	ctx context.Context, taskMetadata *state.TaskResponse, nsenterPrefix string, interfaceName string,
+	flowsPercent int, netemArguments string,
+) error {
+	// Add netem fault to <flowsPercent> third layer bands, this allows to then filter traffic by flow into the 100
+	// third layer bands resulting in <flowsPercent> flows being impacted.
+	for i := range flowsPercent {
```

Comment:
IIUC, depending on the `flowsPercent` value, which can be `100` at max, we could be executing 100 commands back to back here. Is that right? 

If that's indeed correct, then that's not something that should happen while responding to a syncronous API from the caller. How do we know that this is simply not going to timeout (from the caller's perspective) in the middle of one of these executions, especially if there's cpu load on the system?